parcels/particle.py

```diff
@@ -277,16 +355,33 @@ def execute(self, pyfunc=AdvectionRK4, time=None, dt=1., timesteps=1,
         # Check if output is required and compute outer leaps
         if output_file is None or output_steps <= 0:
             output_steps = timesteps
-        timeleaps = int(timesteps / output_steps)
+        timeleaps = int(timesteps / output_steps)   # Number of output points
         # Execute kernel in sub-stepping intervals (leaps)
         current = time or self.grid.time[0]
-        for _ in range(timeleaps):
-            self.kernel.execute(self, output_steps, current, dt)
-            current += output_steps * dt
-            if output_file:
-                output_file.write(self, current)
+        if self.kernel.funcname == 'AdvectionRK45UpdateP' or\
```

Comment:
Wouldn't it make more sense to use `self.kernel.funcname.find('RK45')` here? Because what happens if a user adds yet another kernel? This seems like a slight hack